webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -5485,9 +5428,95 @@ var userOverride = {
+                            var regexForm = $mck_text_box.data(
+                                'validation'
+                            );
+                            regexForm = new RegExp(regexForm);
+                            //If the input does not match the regex validation then show an error message for 2 seconds.
+                            if (!regexForm.test(message)) {
+                                $applozic('#mck-form-field-error-alert').html($mck_text_box.data('errorMessage'));
+                                $applozic('#mck-form-field-error-alert-box')
+                                    .removeClass('n-vis')
+                                    .addClass('vis');
+                                setTimeout(function () {
+                                    $applozic('#mck-form-field-error-alert-box')
+                                        .removeClass('vis')
+                                        .addClass('n-vis');
+                                }, 2000);
+                                return false;
+                            }
+                        }
+                        if($mck_text_box.data('updateUserDetails')){
+                            // If the field is a form field and the user details need to be updated then update the user details
+                            var fieldVal = $mck_text_box.data('field');
+                            var userUpdateField = {};
+                            userUpdateField[fieldVal] = message;
+                            if (
+                                $mck_text_box.data('fieldType') === 'EMAIL' ||
+                                $mck_text_box.data('fieldType') === 'NAME' ||
+                                $mck_text_box.data('fieldType') ===
+                                    'PHONE_NUMBER'
+                            ) {
+                                mckContactService.updateUser({
+                                    data: userUpdateField,
+                                    success: function () {
+                                        sendMessageWhenFieldType();
+                                    },
+                                    error: function () {
+                                        sendMessageWhenFieldType();
+                                    }
+                                });
+                            } else {
+                                mckContactService.updateUser({
+                                    data: { metadata: userUpdateField },
+                                    success: function () {
+                                        sendMessageWhenFieldType();
+                                    },
+                                    error: function () {
+                                        sendMessageWhenFieldType();
+                                    }
+                                });
+                            }
+                        }
```

Comment:
We can improve this logic.
1. you can use switch case for $mck_text_box.data('fieldType')
2. mckContactService.updateUser() is repeating. the data, which is being sent, is different in both places. We can avoid duplication.